test/functional/wallet_encryption.py

```diff
@@ -26,71 +25,72 @@ def run_test(self):
 
-        # Test walletlock
-        self.nodes[0].walletpassphrase(passphrase, 84600)
+        self.log.info("Test walletlock")
+        self.nodes[0].walletpassphrase(passphrase=passphrase, timeout=84600)
         sig = self.nodes[0].signmessage(address, msg)
         assert self.nodes[0].verifymessage(address, sig, msg)
         self.nodes[0].walletlock()
         assert_raises_rpc_error(-13, "Please enter the wallet passphrase with walletpassphrase first", self.nodes[0].signmessage, address, msg)
 
-        # Test passphrase changes
+        self.log.info("Test walletpassphrasechange")
         self.nodes[0].walletpassphrasechange(passphrase, passphrase2)
         assert_raises_rpc_error(-14, "wallet passphrase entered was incorrect", self.nodes[0].walletpassphrase, passphrase, 10)
-        self.nodes[0].walletpassphrase(passphrase2, 10)
+        self.nodes[0].walletpassphrase(passphrase=passphrase2, timeout=10)
         sig = self.nodes[0].signmessage(address, msg)
         assert self.nodes[0].verifymessage(address, sig, msg)
         self.nodes[0].walletlock()
 
-        # Test timeout bounds
+        self.log.info("Test walletpassphrase with an invalid negative timeout and a valid zero timeout")
         assert_raises_rpc_error(-8, "Timeout cannot be negative.", self.nodes[0].walletpassphrase, passphrase2, -10)
-
-        self.log.info('Check a timeout less than the limit')
-        MAX_VALUE = 100000000
-        expected_time = int(time.time()) + MAX_VALUE - 600
-        self.nodes[0].walletpassphrase(passphrase2, MAX_VALUE - 600)
-        # give buffer for walletpassphrase, since it iterates over all encrypted keys
-        expected_time_with_buffer = time.time() + MAX_VALUE - 600
-        actual_time = self.nodes[0].getwalletinfo()['unlocked_until']
-        assert_greater_than_or_equal(actual_time, expected_time)
-        assert_greater_than(expected_time_with_buffer, actual_time)
-
-        self.log.info('Check a timeout greater than the limit')
-        expected_time = int(time.time()) + MAX_VALUE - 1
-        self.nodes[0].walletpassphrase(passphrase2, MAX_VALUE + 1000)
-        expected_time_with_buffer = time.time() + MAX_VALUE
-        actual_time = self.nodes[0].getwalletinfo()['unlocked_until']
-        assert_greater_than_or_equal(actual_time, expected_time)
-        assert_greater_than(expected_time_with_buffer, actual_time)
+        self.nodes[0].walletpassphrase(passphrase=passphrase2, timeout=0)
+
+        # Maximum time to keep the decryption key, in seconds (~3 years); see MAX_SLEEP_TIME
+        timeout_limit = 100_000_000
+
+        self.log.info("Test walletpassphrase with a timeout less than the limit")
+        timeout = timeout_limit - 600
+        before = floor(time.time()) + timeout
+        self.nodes[0].walletpassphrase(passphrase2, timeout)
+        after = ceil(time.time()) + timeout
+        actual = self.nodes[0].getwalletinfo()['unlocked_until']
+        assert before <= actual <= after
```

Comment:
Would this print the values if the assertion fails?

```
>>> a=1
>>> b=2
>>> c=3
>>> assert a <= b <= c
>>> 
>>> c=0
>>> assert a <= b <= c
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> 
```

If not, then maybe revert back to using `assert_greater_than()` because seeing the values in CI logs is so precious in debugging this.

Reply:
Looking.

Reply:
Good point. Updated the assertion error to print i.e. `AssertionError: walletpassphrase "unlocked_until" value of 1760141911 not in expected range of 1760141909..1760141910`